### PR TITLE
Playlist watchlater

### PR DESCRIPTION
**Pull request by DivyanshSareen**
implemented:
- watch later
    - add vid to watch later
    - remove vid from watch later
- playlist management
    - add new playlist
    - delete playlist
    - add video to playlist
    - remove video from playlist

### CODE REVIEW

File: src/components/Cards/WatchlaterCard.jsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      <Link
+        to={`/video/${video._id}`}
+        onMouseEnter={() => setMouseOnCard(true)}
+        onMouseLeave={() => setMouseOnCard(false)}
+        className='card-img resp-img'>
+        <img
+          src={require(`../../assets/${video.image}`)}
+          alt='thumbnail'
+          style={{
+            opacity: mouseOnCard === true ? "50%" : "100%",
```
Comment (harshalini) [inline src/components/Cards/WatchlaterCard.jsx:17]: Liked the logic for style here

File: src/styles/modal.css [hunk 2]
```diff
@@ -0,0 +1,20 @@
+.modal {
+    display: flex;
+    width: 100%;
+    height: 100%;
+    justify-content: center;
+    align-items: center;
+    background-color: rgb(0, 0, 0);
+    background-color: rgba(43, 43, 43, 0.9);
```
Comment (harshalini) [inline src/styles/modal.css:8]: Remove first background-color as it is getting overridden

File: src/routes/VideoPage.jsx [hunk 3]
```diff
@@ -66,7 +79,14 @@ const VideoPage = () => {
             </div>
             <div className='video-actions'>
               <RequiresAuth>
-                <i className='action-icon fa-regular fa-clock fa-lg'></i>
+                <i
+                  className='action-icon fa-regular fa-clock fa-lg'
+                  style={{ opacity: `${isInWatchlater ? "100%" : "50%"}` }}
```
Comment (harshalini) [inline src/routes/VideoPage.jsx:84]: You can rather set opacity as 1 and 0.5,  as I think it is not supported in all browsers in percentage

File: src/components/Cards/PlaylistItem.jsx [hunk 4]
```diff
@@ -1,12 +1,14 @@
-const PlaylistItem = ({ video }) => {
+const PlaylistItem = ({ playlistId, video, removeVideoFromPlaylist }) => {
   return (
     <div className='card playlist-item'>
       <div className='card-content'>
         <div className='card-head'>
           <div className='card-title h4'>{video.title}</div>
           <div className='card-subtitle sub-title2'>{video.creator}</div>
```
Comment (harshalini) [inline src/components/Cards/PlaylistItem.jsx:7]: Suggestion: you can destructure creator and title from video

**Review by harshalini — APPROVED** [review]
Readable and well-formatted code, just left some suggestions, good to merge this PR

**Review by yashraj2312 — APPROVED** [review]
Looks good to merge

File: src/styles/modal.css (hunk 2 above)
Comment (DivyanshSareen) [reply src/styles/modal.css:8]: I added it as a fallback to second one

File: src/routes/VideoPage.jsx (hunk 3 above)
Comment (DivyanshSareen) [reply src/routes/VideoPage.jsx:84]: Really appreciate the reviews, will definitely solve the issues in a later commit

**Pull request merged**